Grep/src/bg/sofia/uni/fmi/mjt/grep/StringSearcher.java

```diff
@@ -0,0 +1,102 @@
+package bg.sofia.uni.fmi.mjt.grep;
+
+import bg.sofia.uni.fmi.mjt.grep.CustomExceptions.*;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.*;
+import java.util.concurrent.*;
+import java.util.stream.Collectors;
+
+// executor shutdown()??
+public class StringSearcher {
+    private String word;
+    private String folderPath;
+    private ExecutorService executor;
+    private String outPutFilePath;
+    private Map<String, Boolean> OPTION_MAP;
+    private InputGrepCommandParser inputGrepCommandParser;
+    private InputCommandValidator inputCommandValidator;
+
+    // close the source when??
+    private InputStream source;
+
+    public StringSearcher() {
+    }
+
+    public List<String> stringSearch(InputStream source) throws Throwable {
+        // add while until quit
+        this.source = source;
+
+        String input = read(this.source);
+
+        inputCommandValidator = new InputCommandValidator();
+        inputCommandValidator.validateRequiredCommand(input);
+
+        inputGrepCommandParser = new InputGrepCommandParser(input.split(" "));
```

Comment:
Make it so, that command parser works even if `getOptions` is not called as first method, otherwise someone else might not use it the same way if that person hasn't seen the implementation. Without looking at the code, I would assume so

Reply:
We can encapsulate all of these operations in command parser class and return Map with appropriate values, directly from, let's say function getCommandData(). Is it ok?